### DISCUSSION TITLE
How to use gpt-4o with version 2024-08-06?

### DISCUSSION THREAD

**Discussion by zch-cc** [community/community#145751] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi Team, I want to use structure output for gpt4o and I think 2024-08-06 support that. My question is how to use it? 
I tried 
```
client = OpenAI(
                base_url="https://models.inference.ai.azure.com",
                api_key=config.LLM_TOKEN)
```
Does not work with error openai.BadRequestError: Error code: 400 - {'error': {'code': 'BadRequest', 'message': 'response_format value as json_schema is enabled only for api versions 2024-08-01-preview and later'}}

Also tried 
```
client = AzureOpenAI(
                base_url="https://models.inference.ai.azure.com",
                api_key=config.LLM_TOKEN, api_version="2024-08-06")
```
Also does not work with 404 error 
httpx HTTP Request: POST https://models.inference.ai.azure.com/deployments/gpt-4o/chat/completions?api-version=2024-08-06 "HTTP/1.1 404 Not Found"

**Comment by kevin-m-kent** [comment]
Hi - try api-version=2024-08-01-preview. This should work

**Comment by KateCatlin** [comment]
Hey @zch-cc did you get it working? If not I can open a bug ticket for you! 

Also I'd love to hear more about what you're using this for if you'd be willing to share :)

**Reply by zch-cc** [reply, marked as answer]
Hi Kate, I got work with 
api_version="2024-10-21" with AzureOpenAI
and  base_url = "https://models.inference.ai.azure.com/openai"
I am using sturcture output but the old api version does not support this

**Reply by KateCatlin** [reply]
Awesome, thanks for letting me know @zch-cc. We're considering how we can better handle needing the API versions for structured outputs. 

Out of curiosity, can I ask what you're using GitHub Models for? If it's open-source or deployed somewhere I'd love to see what you've built with it!
